nestkernel/event.h

```diff
@@ -1062,6 +1075,450 @@ class GapJunctionEvent : public SecondaryEvent
   double get_coeffvalue( std::vector< unsigned int >::iterator& pos );
 };
 
+/**
+ * Event for rate model connections without delay.
+ */
+class RateNeuronEvent : public SecondaryEvent
```

Comment:
@heplesser Yes, there is actually a lot of overlap and I am not particular happy about that. I already thought about how to reduce this overlap, but this is not that easy. Each class contains important individual static member variables, which are used in almost every function. 

Let me explain in more detail: `GapJunctionEvent`,`RateNeuronEvent`,`DelayRateNeuronEvent` and `DiffusionEvent` all contain around 200 lines of code, but are more or less identical classes: All of them store arbitrary data of type `std::vector<double>`. Therefore it would be much better to have one base class (e.g. `DoubleVectorEvent`) of which all the other classes would be derived.

I could think of two approaches to achieve this:

**1. use derived classes**
The problem here is that derived classes do not have their own instance of the static variables. One possible solution ( http://stackoverflow.com/a/1390922 ) is to add a virtual function `type& get_static()` to the base class and call the static variable only with this function. Then each derived class can define its own static variables and by overwriting the virtual `get_static` function(s) they are also accessible for the functions of the base class.

This approach however does not work for us, as we also have a static function `static void set_syn_id( const synindex synid )` in event classes (and in the potential base class) which calls one of the static variables. Calling this static variable with the `get_static` function yields an `cannot call member function without object`-error - which means that `get_static` would have to be static as well, which is not possible as it is a virtual function.

2. **use a template class**
Employ a `template <typename something> class DoubleVectorEvent` with unused (but unique) template parameters. In this case each template class has its own static members. So one could define the events like `typedef DoubleVectorEvent<GapJunction> GapJunctionEvent` which would be nice as well. One (however solvable) problem here is the initialization of the static members. A much bigger problem is the operator `void DoubleVectorEvent::operator()()` which has to stay in `event.cpp` due to dependencies to `node.h`. A template definition of the class would however require to move the operator to `event.h`.

So for now we saw no solution to reduce the overhead here. If you have any suggestion they are of course very welcome!